embedded-graphics/src/image/mod.rs

```diff
@@ -1,33 +1,262 @@
+    /// assert_eq!(image_moved.top_left(), Point::new(10, 20));
+    /// ```
+    fn translate(&self, by: Point) -> Self {
+        Self {
+            image_data: self.image_data,
+            offset: self.offset + by,
+            c: PhantomData,
+        }
+    }
+
+    /// Translate the image by a given delta, modifying the original object
+    ///
+    /// # Examples
+    ///
+    /// ## Move an image around
+    ///
+    /// This examples moves a 4x4 black and white image by `(10, 20)` pixels by mutating the
+    /// original image
+    ///
+    /// ```rust
+    /// use embedded_graphics::{
+    ///     geometry::Point,
+    ///     image::{Image, ImageRaw},
+    ///     pixelcolor::BinaryColor,
+    ///     prelude::*,
+    /// };
+    ///
+    /// let image: ImageRaw<BinaryColor> = ImageRaw::new(&[0xff, 0x00, 0xff, 0x00], 4, 4);
+    ///
+    /// let mut image = Image::new(&image, Point::zero());
+    ///
+    /// image.translate_mut(Point::new(10, 20));
+    ///
+    /// assert_eq!(image.top_left(), Point::new(10, 20));
+    /// ```
+    fn translate_mut(&mut self, by: Point) -> &mut Self {
+        self.offset += by;
+
+        self
+    }
+}
+
+impl<'a, 'b, I, C> Drawable<C> for &'a Image<'b, I, C>
+where
+    &'b I: IntoPixelIter<C>,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_iter(
+            self.image_data
+                .pixel_iter()
+                .map(|p| Pixel(p.0 + self.offset, p.1)),
+        )
+    }
+}
+
+impl<'a, I, C> Dimensions for Image<'a, I, C>
+where
+    I: ImageDimensions,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    fn top_left(&self) -> Point {
+        self.offset
+    }
+
+    fn bottom_right(&self) -> Point {
+        self.top_left() + self.size()
+    }
+
+    fn size(&self) -> Size {
+        Size::new(self.image_data.width(), self.image_data.height())
+    }
+}
+
+impl<'a, 'b, I, C> IntoIterator for &'a Image<'b, I, C>
+where
+    &'b I: IntoPixelIter<C>,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    type Item = Pixel<C>;
+    type IntoIter = ImageIterator<'a, 'b, I, C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        ImageIterator {
+            it: self.image_data.pixel_iter(),
+            image: self,
+        }
+    }
+}
+
+/// Pixel iterator over `Image` objects
+pub struct ImageIterator<'a, 'b, I, C>
+where
+    &'b I: IntoPixelIter<C>,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    image: &'a Image<'b, I, C>,
+    it: <&'b I as IntoPixelIter<C>>::PixelIterator,
+}
+
+impl<'a, 'b, I, C> fmt::Debug for ImageIterator<'a, 'b, I, C>
+where
+    &'b I: IntoPixelIter<C>,
+    C: PixelColor + From<<C as PixelColor>::Raw>,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        // The Debug derive didn't work anymore without additional trait bounds
+        // TODO: add fields
+        f.debug_struct("ImageIterator").finish()
```

Comment:
I did add this comment to my lifetime fix. But I think this should be fixed and the comment should be removed before this is merged.